src/target/stm32f4.c

```diff
@@ -432,12 +432,12 @@ static int stm32f4_flash_erase(target_flash_s *f, target_addr_t addr, size_t len
 	sr = target_mem_read32(t, FLASH_SR);
 	if(sr & SR_ERROR_MASK) {
 		DEBUG_WARN("stm32f4 flash erase: sr error: 0x%" PRIx32 "\n", sr);
-		return -1;
+		return false;
```

Comment:
Can the returns here be elided? looks like it could be `return !(sr & SR_ERROR_MASK);`